server/game/player.js

```diff
@@ -603,6 +603,7 @@ class Player extends GameObject {
                 }
             });
         } else {
+            this.modifyAmber(-modifiedCost);
```

Comment:
There's a lot of duplicated code here, can you refactor this into a function while you're here?